docs/src/App.js

```diff
@@ -0,0 +1,97 @@
+import './App.css';
+import {useState} from "react";
+import DeleteCompletedAlert from "./DeleteCompletedAlert";
+import ListItems from "./ListItems";
+import {generateUniqueID} from "web-vitals/dist/modules/lib/generateUniqueID";
+
+function App(props) {
+    const [items, setItems] = useState(props.initialData);
+    const [completedItems, setCompletedItems] = useState([]);
+    const [showDeleteAlert, setShowDeleteAlert] = useState(false);
+    // const deletedItems = [];
+    const [totalItems, setTotalItems] = useState(props.initialData);
+
+    function handleEditItem(itemId, value, field) {
+        setItems(
+            items.map(
+                (item) => (item.id === itemId ? {...item, [field]: value} : item)
+            )
+        );
+        setTotalItems(items)
+    }
+
+    function handleAddItem(e) {
+        if (e.key === 'Enter') {
+            const newItem = {id: generateUniqueID(), value: e.target.value, completed: false}; // add a boolean for completed
+            setItems([...items, newItem]);
+            totalItems.push(newItem);
+            e.target.value = "";
+        }
+    }
+
+    // function handleDeleteItem() {
+    //
+    // }
+    //
+    // function handleUndoDelete() {
+    //
+    // }
+
+    function handleDeleteCompleted() {
+        setTotalItems(totalItems.filter(i => !completedItems.includes(i.id)));
+        setItems(items.filter(i => !completedItems.includes(i.id)));
+        setCompletedItems([]);
+        setShowDeleteAlert(!showDeleteAlert);
+        console.log(items);
+        console.log(totalItems);
+    }
+
+    function handleToggleCompleted(e) {
+        console.log(totalItems);
+        if (e.target.checked) {
+            setItems(items.filter(i => !completedItems.includes(i.id)));
+        } else {
+            setItems(totalItems);
+        }
+    }
+
+    function handleCompletedItems(item) {
+        setCompletedItems([...completedItems, item.id]);
+        item.completed = true;
```

Comment:
item is oen of the elements of a state variable.  You can't change anything within a state variable directly. You must use the set method.  Thus, you must create a new item and change the value in there.